`format_<prediction type>()` functions only format when necessary

hfrick opened the pull request:
closes  #263

The main part was done in #268 but since the issue was still open, this PR now extends it to newer prediction types.

EmilHvitfeldt reviewed (commented):
I like the consistancy upgrade 🙌 

Are these similar enough that we can have a helper function do it?

```r
this_would_need_a_name <- function(x, col_name) {
  if (isTRUE(ncol(x) > 1) | is.data.frame(x)) {
    x <- as_tibble(x, .name_repair = "minimal")
    if (!any(grepl(paste0("^\\", col_name), names(x)))) {
      names(x) <- col_name
    }
  } else {
    x <- tibble(unname(x))
    names(x) <- col_name
    x
  }
}
```

hfrick commented:
Good thought - I'd say so, yes 🙌  I've introduced a helper here and added a follow-up issue #927. We might get to a general `format_parsnip()` helper but I didn't want to overload this PR.

EmilHvitfeldt commented:
Very nice! I think this is a good improvement
